auto3dseg/algorithm_templates/dints/scripts/train.py

```diff
@@ -148,30 +157,30 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
         cache_rate=train_cache_rate,
         hash_as_key=True,
         num_workers=parser.get_parsed_content("training#num_cache_workers"),
-        progress=False,
+        progress=True,
     )
     val_ds = monai.data.CacheDataset(
         data=val_files,
         transform=val_transforms,
         cache_rate=validate_cache_rate,
         hash_as_key=True,
         num_workers=parser.get_parsed_content("training#num_cache_workers"),
-        progress=False,
+        progress=True,
     )
 
     train_loader = DataLoader(
         train_ds,
         num_workers=parser.get_parsed_content("training#num_workers"),
         batch_size=num_images_per_batch,
         shuffle=True,
+        persistent_workers=True,
+        pin_memory=True,
     )
-    val_loader = DataLoader(val_ds, num_workers=0, batch_size=1, shuffle=False)
-
-    device = (
-        torch.device(f"cuda:{os.environ['LOCAL_RANK']}")
-        if world_size > 1
-        else torch.device("cuda:0")
+    val_loader = DataLoader(
```

Comment:
Maybe we should make `num_workers`, `batch_size`, `shuffle`, `persistent_workers`, `pin_memory` all configurable by configs to adapt different environments.

Reply:
I added `num_workers` for validation data loader. For other options, they either cannot be changed (like batch size), or do not make any significant difference when the user changes them.